### ISSUE TITLE
[B] jqueryui-datepicker fails on Google translated pages

### ISSUE THREAD

**Issue by githubertus** [opensource-socialnetwork/opensource-socialnetwork#2036]
I was wondering already some weeks which way people managed to register with a birthdate like `NaN/NaN/NaN`

Finally, I found the explanation here: https://bugs.jqueryui.com/ticket/8266

And actually you'll get it on an English page translated to German ...
![google-translate-datepicker](https://user-images.githubusercontent.com/9904364/144744855-7a4f952f-0975-4129-9856-364a1b0f8082.png)

Possible fix: https://sarathlal.com/stop-google-from-translating-datepicker-input-field/

**Comment by lianglee** [comment]
I think we should also do validation in backend

**Comment by michieal** [comment]
@githubertus I got it translating the page from english to english. I also tried adding the "notranslate" class to the input, and it didn't change the NaN/NaN/NaN from the field. Was curious to know if you got it to work at all?

**Comment by lianglee** [comment]
@michieal depending on what version of OSSN you are running and did you verified if your birthdate field have notranslate class?

**Comment by michieal** [comment]
Ossn 6.1; And I specifically added (and checked in the page source) the notranslate class.

**Comment by lianglee** [comment]
@michieal You should check if the change effects the actual element you can use browser view source code to see.

**Comment by lianglee** [comment]
Not sure then what is cause if no translate class is there. Need to check

**Comment by michieal** [comment]
The "notranslate" class was added to the text input / textarea for the Birthdate field (that's read only). It filled it with NaN's when translated. 
So, not sure - I followed the above, and I am pretty sure that I did it right.

**Comment by githubertus** [comment]
Are we talking about THIS site?
![nan](https://user-images.githubusercontent.com/9904364/158342264-c54965f0-53c6-418c-8fd4-b3734724b723.png)

**Comment by michieal** [comment]
@githubertus -- I used my beta site, not my live site.

**Comment by michieal** [comment]
I added this to the profile js:
`		var args = Ossn.call_hook('profile', 'birthdate:input', null, datepick_args);	
		$("input[name='birthdate']").datepicker(args);
        $("input[name='birthdate']").addClass('notranslate');
`

And I also tried this in ossn.lib.users:
`        'text' => array(
            array(
                'name' => 'birthdate',
                'params' => array(
                    'readonly' => true,
                    'class' => 'notranslate', // this line here.
                )
            )
`

**Comment by githubertus** [comment]
Ok, and your beta site is a clean Ossn 6.x without any enabled extras?

**Comment by michieal** [comment]
The "notranslate" class was added to the text input / textarea for the Birthdate field (that's read only). It filled it with NaN's when translated. 
So, not sure - I followed the above, and I am pretty sure that I did it right. 

> Ok, and your beta site is a clean Ossn 6.x without any enabled extras?

uhm, not really. my beta site is the last testing zone before making things live. I dev on a vm, but I found out the hard way that there are differences between the vm and it being in the wild; so I made a mostly wild back up copy of the live site to test it one last time before making it live. Only I have access to the beta, as I don't want complete screw ups to be seen... kinda breaks the mystique if others see abject fails on a daily basis. lol
